homeassistant/components/plugwise/config_flow.py

```diff
@@ -154,42 +170,92 @@ def is_matching(self, other_flow: Self) -> bool:
+            errors[CONF_BASE] = "unsupported"
+        except Exception:  # noqa: BLE001
+            errors[CONF_BASE] = "unknown"
+        else:
+            await self.async_set_unique_id(
+                api.smile_hostname or api.gateway_id, raise_on_progress=False
+            )
+            return (api, errors)
+        return (None, errors)
+
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Handle the initial step when using network/gateway setups."""
         errors: dict[str, str] = {}
 
-        if user_input is not None:
-            if self.discovery_info:
-                user_input[CONF_HOST] = self.discovery_info.host
-                user_input[CONF_PORT] = self.discovery_info.port
-                user_input[CONF_USERNAME] = self._username
+        if not user_input:
+            return self.async_show_form(
+                step_id=SOURCE_USER,
+                data_schema=base_schema(self.discovery_info),
+                errors=errors,
+            )
 
-            try:
-                api = await validate_input(self.hass, user_input)
-            except ConnectionFailedError:
-                errors[CONF_BASE] = "cannot_connect"
-            except InvalidAuthentication:
-                errors[CONF_BASE] = "invalid_auth"
-            except InvalidSetupError:
-                errors[CONF_BASE] = "invalid_setup"
-            except (InvalidXMLError, ResponseError):
-                errors[CONF_BASE] = "response_error"
-            except UnsupportedDeviceError:
-                errors[CONF_BASE] = "unsupported"
-            except Exception:  # noqa: BLE001
-                errors[CONF_BASE] = "unknown"
-            else:
-                await self.async_set_unique_id(
-                    api.smile_hostname or api.gateway_id, raise_on_progress=False
-                )
-                self._abort_if_unique_id_configured()
+        if self.discovery_info:
+            user_input[CONF_HOST] = self.discovery_info.host
+            user_input[CONF_PORT] = self.discovery_info.port
+            user_input[CONF_USERNAME] = self._username
 
-                return self.async_create_entry(title=api.smile_name, data=user_input)
+        api, errors = await self._verify_connection(user_input)
+        if not errors and api:
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(title=api.smile_name, data=user_input)
 
         return self.async_show_form(
             step_id=SOURCE_USER,
             data_schema=base_schema(self.discovery_info),
             errors=errors,
         )
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle reconfiguration of the integration."""
+        errors: dict[str, str] = {}
+
+        reconfigure_entry = self._get_reconfigure_entry()
+
+        if user_input:
+            user_input = {
```

Comment:
Maybe rework the logic to `if not user_input:` similar to the structure in `async_step_user()`, does that make sense?

Reply:
It needed rework, more closer to the original, but the `user` step assumes (unless zeroconf) ran without data while the `reconfigure` will by default re-use data. Even though they are 'if not vs not' in the forthcoming commit, I think they'll make more sense this way.